pkg/controller/restore_finalizer_controller.go

```diff
@@ -311,7 +311,7 @@ func (ctx *finalizerContext) patchDynamicPVWithVolumeInfo() (errs results.Result
 					// We are handling a common but specific scenario where a PVC is in a pending state and uses a storage class with
 					// VolumeBindingMode set to WaitForFirstConsumer. In this case, the PV patch step is skipped to avoid
 					// failures due to the PVC not being bound, which could cause a timeout and result in a failed restore.
-					if pvc != nil && pvc.Status.Phase == v1.ClaimPending {
```

Comment:
should we do pvc.status == nil?
That could be nil right?

Reply:
https://github.com/vmware-tanzu/velero/pull/8071#discussion_r1700776388

Tried and got an error